### PR TITLE
Adding Next.JS QSG for Asgardeo and IS

### PR DESCRIPTION
**Pull request by sagara-gunathunga**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+export const { handlers, signIn, signOut, auth } = NextAuth({
+  providers: [Asgardeo({
+    issuer: process.env.AUTH_ASGARDEO_ISSUER
+  })],
+})
+
+```
+
+
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:163]:
```suggestion
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+Create a Route Handler file in the `src/app/api/auth/[...nextauth]/route.ts` location. 
+
+```bash
+mkdir -p src/app/api/auth/\[...nextauth\]
+
+touch mkdir -p src/app/api/auth/\[...nextauth\]/route.ts
+
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:171]:
```suggestion
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+```
+
+!!! Note
+    The directory `src/app/api/auth/[...nextauth]/route.ts` in a Next.js project is used to define a dynamic API route for handling authentication. The `[...nextauth]` is a catch-all route that processes multiple authentication-related requests such as sign-in, sign-out, and session management. The route.ts file specifies the logic for these operations, typically by exporting handlers for HTTP methods like GET and POST. This setup centralizes authentication logic, supports OAuth providers like Google or GitHub, and integrates seamlessly into Next.js applications for secure and scalable authentication workflows.
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:175]: The directory `src/app/api/auth/[...nextauth]/route.ts` -> The file `src/app/api/auth/[...nextauth]/route.ts``

Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:175]:
```suggestion
    The directory `src/app/api/auth/[...nextauth]/route.ts` in a Next.js project is used to define a dynamic API route for handling authentication. The `[...nextauth]` is a catch-all route that processes multiple authentication-related requests such as sign-in, sign-out, and session management. The `route.ts` file specifies the logic for these operations, typically by exporting handlers for HTTP methods like GET and POST. This setup centralizes authentication logic, supports OAuth providers like Google or GitHub, and integrates seamlessly into Next.js applications for secure and scalable authentication workflows.
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+
+Update the `src/app/api/auth/[...nextauth]/route.ts` file with the following code. 
+
+```javascript title="route.ts"
+import { handlers } from "@/auth" 
+export const { GET, POST } = handlers
+```
+
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:184]:
```suggestion
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+            <form
+              action={async () => {
+                "use server"
+                await signOut()
+              }}
+            >
+              <button type="submit">Sign Out</button>
+            </form>
+          </>
+
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:305]:
```suggestion
<>
  <p> You are now signed in!</p>
  <p> hello {session.user?.username}</p>
  <form
    action={async () => {
      "use server"
      await signOut()
    }}
  >
    <button type="submit">Sign Out</button>
  </form>
</>
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+...
+
+```
+
+
+
```
Comment (pavinduLakshan) [inline en/asgardeo/docs/quick-starts/nextjs.md:311]:
```suggestion
```

File: en/asgardeo/docs/quick-starts/nextjs.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+  - About 15 minutes
+  - <a href="{{ base_path }}/get-started/create-asgardeo-account/">Asgardeo account</a>
+  - Install a JS package manager
+  - A favorite text editor or IDE
+# source_code: <a href="" target="_blank" class="github-icon">Next.js App Sample</a>
+whats_next:
+  # - Try out <a href="{{ base_path }}/complete-guides/react/introduction/" target="_blank">{{ product_name }} complete React guide</a>
+  # - Try out {{product_name}} user onboarding complete guide for React
+  # - Read security best practices for React app guide
+---
```
[the remark this answers is not in the document]
Comment (sagara-gunathunga) [reply en/asgardeo/docs/quick-starts/nextjs.md:20]: Yes at this stage we need to keep them commented out until we get the content for the links.

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**